symbolic-debuginfo/src/pe.rs

```diff
@@ -357,54 +387,26 @@ impl<'data, 'object> Iterator for PeSymbolIterator<'data, 'object> {
     }
 }
 
-/// Debug session for PE objects.
-///
-/// Since debug information in PE containers is not supported, this session consists of NoOps and
-/// always returns empty results.
-#[derive(Debug)]
-pub struct PeDebugSession<'data> {
-    _ph: PhantomData<&'data ()>,
-}
-
-impl<'data> PeDebugSession<'data> {
-    /// Returns an iterator over all functions in this debug file.
-    pub fn functions(&self) -> PeFunctionIterator<'_> {
-        std::iter::empty()
-    }
-
-    /// Returns an iterator over all source files referenced by this debug file.
-    pub fn files(&self) -> PeFileIterator<'_> {
-        std::iter::empty()
-    }
-
-    /// Looks up a file's source contents by its full canonicalized path.
-    ///
-    /// The given path must be canonicalized.
-    pub fn source_by_path(&self, _path: &str) -> Result<Option<Cow<'_, str>>, PeError> {
-        Ok(None)
+impl<'data> Dwarf<'data> for PeObject<'data> {
+    fn endianity(&self) -> RunTimeEndian {
+        // According to https://reverseengineering.stackexchange.com/questions/17922/determining-endianness-of-pe-files-windows-on-arm,
+        // the only known platform running PE's with big-endian code is the Xbox360. Probably not worth handling.
+        return RunTimeEndian::Little;
+    }
+
+    fn raw_section(&self, name: &str) -> Option<DwarfSection<'data>> {
+        // Name is given without leading "."
+        let sect = self.section(&format!(".{}", name))?;
+        // TODO: Proper error handling if conversion fails
+        let dwarf_data: &'data [u8] = &self.data[sect.pointer_to_raw_data.try_into().unwrap()..]
+            [..sect.virtual_size.try_into().unwrap()];
```

Comment:
I believe we can just use `as` casting for the number values here.
However I would very much prefer the use of `.get()` to be safe against out of bounds access. Slice syntax panics if one of these is out of bounds. As we use "untrusted" values coming from the section table, we should be defensive here.

Reply:
Makes sense, I've implemented your suggestions.